UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/MetaTileMap.cs

```diff
@@ -1272,6 +1273,12 @@ public void RemoveTileWithlayer(Vector3Int position, LayerType refLayer)
 
 				if (TileLcation != null)
 				{
+					if (TileLcation.Tile is BasicTile tile && tile.SoundOnDestroy.Count > 0)
+					{
+						// Because this function is called through a network message, the sound does not have to
+						// be played through the network, as all clients will receive the network message.
+						SoundManager.PlayAtPosition(tile.SoundOnDestroy.RandomElement(), position);
+					}
```

Comment:
Dont think this is the best place to put this as it would play for any type of removing/changing a tile, check out TilemapDamage.cs around line 96